### PR TITLE
cfssl scan: Initial API endpoint

### PR DESCRIPTION
**Pull request by jacobhaven**
New API endpoint for cfssl scan.

### CODE REVIEW

File: api/scan/scan.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+		return errors.NewBadRequest(err)
+	}
+
+	responses := make(map[string][]FamilyResponse)
+	for _, host := range req.Hosts {
+		if _, _, err := net.SplitHostPort(host); err != nil {
+			host = net.JoinHostPort(host, "443")
+		}
+
+		for _, family := range scan.AllFamilies {
```
Comment (grittygrease) [inline api/scan/scan.go]: This has too many levels of depth for a main function. Can you extract the regular expression logic into its own function and add test cases?

File: doc/api.txt [hunk 2]
```diff
@@ -314,6 +314,37 @@ Example:
...
+Example:
+
+    scan.json:
+    {
+        "hosts": [
+            "example.com",
+            "cloudflare.com"
+            "localhost:44433"
+        ]
+        family: "Connectivity"
```
Comment (grittygrease) [inline doc/api.txt]: Can you add the list of families, or a way to list families here?

File: api/scan/scan.go (hunk 1 above)
Comment (jacobhaven) [inline api/scan/scan.go]: Yeah, I'd agree. It's was also duplicated in the cli package. I merged them into one scan/output loop (though this breaks realtime cli output...) I might eventually make it spawn a goroutine and return response structs (easily JSON marshalable) one at a time. Having realtime results (non-buffered or flushing http response) using net/http ResponseWriter is more complicated, but possible.

File: doc/api.txt (hunk 2 above)
Comment (jacobhaven) [inline doc/api.txt]: I added a scaninfo endpoint that is just a static list of the families/scanners, but every result natively contains a list of all the families/scanners (and their descriptions) that were run.

**Comment by jacobhaven** [comment]
I set up a test server [here](http://cfssl.jhaven.me:8888/api/v1/cfssl/scan?host=cloudflare.com)

[This site](https://www.jsoneditoronline.org/?url=http://cfssl.jhaven.me:8888/api/v1/cfssl/scan?host=cloudflare.com) is pretty nice for inspecting the JSON.

**Comment by grittygrease** [comment]
Looks good.

**Comment by lziest** [comment]
LGTM

**Comment by kisom** [comment]
LGTM.

**Pull request merged**